### DISCUSSION TITLE
Dry runs / noDeploy cli option disappeared in 2.36.0

### DISCUSSION THREAD

**Discussion by Trolleymusic** [serverless/serverless#9389] [Q&A] [answered]
Hello, I've got a question as I was about to file a bug report - but am wondering if it is a bug, or if there's an intention behind the change:

So this pull from 9 April #9281 Includes the line

> Remove handling of internal `noDeploy` option

And the [release information for 2.36.0](https://github.com/serverless/serverless/blob/1626761b31655b459b05759fca17c279c61b41a2/CHANGELOG.md) includes the line:

> - Remove support for internal `noDeploy` option ([#9281](https://github.com/serverless/serverless/pull/9281)) ([688d09b](https://github.com/serverless/serverless/commit/688d09b1f7f28f8b1656d3f7f31c6c2765a01b9d)) ([Mariusz Nowak](https://github.com/medikoo))

And ever since then, the `--noDeploy` cli flag does not work. I can't do a dry-run deployment.

This contradicts the documentation, but the documentation could be out of date (it was only updated to include `noDeploy` in February #8922). I would be happy to open a PR to update the docs, but I don't know what the intention was behind removing `noDeploy`.

Is there an alternative way of running a dry-run deploy from the command line?

Also, apologies for pulling you into a discussion @medikoo , but I wonder if you have any insight since it was your commit I'm referencing above?

Thanks

**Comment by medikoo** [comment, marked as answer]
@Trolleymusic great thanks for that input. When doing that change I overlooked that this option is documented in our documentation.

This option (1) wasn't registered as a command option internally, so was not listed in list of supported options in `sls deploy --help` and (2) internally was used just in a single test. Seeing that I assumed that it was created simply for purpose of this test (which I was refactoring then), so I decided to remove it as it didn't feel clean.

> Is there an alternative way of running a dry-run deploy from the command line?

What is the use case for this command on your side?

**Reply by Trolleymusic** [reply]
@medikoo - no stress, it's a huge project and I didn't think you did it for anything other than good reasons 🙂  

> What is the use case for this command on your side?

I'm using it to validate the `serverless.yml` and that variables/dependencies (local or pulled in from AWS SSM) are in place.

I use it often locally while developing to confirm I'm configuring things correctly/as expected, but also use it in CI pipelines - running on opening/updating pull requests.

I'm happy to use another command (I've seen a reference to `shouldNotDeploy`, but cannot see how to use it) - but it makes sense to me to be able to choose to do a dry run (eg: the aws-cli lambda command `update-function-code`  has a `--dry-run` option). I'd rather a failed PR build/check than a failed deploy because of invalid yaml/misconfiguration showing up **after** merging to master/dev/etc.

**Reply by medikoo** [reply]
@Trolleymusic `--dry-run` is an interesting functionality, and I think it'll be good to bring at some point (feel free to open request for that in issues).

 What `--notDeploy` did was actually equivalent to doing `sls package` instead of `sls deploy`, so it technically aborted any deploy logic.  I'd imagine `--dry-run`  should go into _deploy_ logic, but not issuing an actual AWS SDK requests.
 
 Anyway I think you can address your use case simply by doing `sls package` or `sls print`

**Reply by Trolleymusic** [reply]
@medikoo thanks! `sls package` seems to cover 99% of what I was checking with `--noDeploy` - I'll roll that out to my repos today and see how it goes.

The remaining 1% is to do with custom domains, but I'll either let that drop, or take that up with the plugin project.

**Reply by medikoo** [reply]
Thanks @Trolleymusic for all the insight, I've prepared the PR with which we will remove it from website documentation to reduce confusion: https://github.com/serverless/serverless/pull/9394

**Reply by TechTeam12** [reply]
The dry-run functionality where it did the deploy logic minus the actual deploy would be very helpful. I'm currently using the package in our CI/CD for testing stages, and it catches 99% of things. However, I think with that dry-run I should be able to catch the 99.9% of things. Is there any update for timeline on that functionality?

**Reply by medikoo** [reply]
@TechTeam12 no, please open a request for it
